### PR TITLE
fix: default map to put the first one if the model has maps.

### PR DESCRIPTION
**Pull request by AlbaLopez**

### CODE REVIEW

File: src/app/app-interactive-map/store/interactive-map.effects.ts [hunk 1]
```diff
@@ -51,10 +51,12 @@ const preferredMapsByModel = {
...
 const preferredSelector = <T>(
-    predicate: (item: T) => boolean,
-    preferredMap?: (item: T) => boolean,
-  ) => (items: T[]): T =>
-  items.find(predicate) || (preferredMap ? items.find(preferredMap) || items[0] : items[0]);
+  predicate: (item: T) => boolean,
+  preferredMap?: (item: T) => boolean,
+  modelHasMaps?: (item: T) => boolean,
+) => (items: T[]): T =>
+  items.find(predicate) || (items.filter(modelHasMaps).length > 0 ? items.filter(modelHasMaps)[0] : (items.find(preferredMap) ||
```
Comment (matyasfodor) [inline src/app/app-interactive-map/store/interactive-map.effects.ts]: `items.filter(modelHasMaps).length > 0 ? items.filter(modelHasMaps)[0]` is the same as `items.find(modelHasMaps)`.

Find returns the first element for which the filtering function returns true, if none fulfills the criteria, returns undefined.

By using `||` if the lefthand operation returns a falsy value, it calculates the right hand.

File: src/app/app-interactive-map/store/interactive-map.effects.ts [hunk 2]
```diff
@@ -36,32 +36,52 @@ const ACTION_OFFSETS = {
...
 const preferredSpecies = preferredSelector((species: types.Species) =>
   preferredSpeciesList.includes(species.name));
 
+const compareModel = (a, b) => a && b && a.model === b.model;
+const compareName = (a, b) => a && b && a.name === b.name;
+
+const selectorCreator = <T>(comparators: ((a: T, b: T) => boolean)[]) => (items: T[], preferredItem: T, defaultMap) => {
+  return comparators.reduce((prev: T[], next: (a: T, b: T) => boolean) => {
+    const filtered = prev.length > 0 ? prev.filter((item) => next(item, preferredItem)) : [];
+    return filtered.length > 0 ? filtered : prev[0] || defaultMap;
```
Comment (AlbaLopez) [inline src/app/app-interactive-map/store/interactive-map.effects.ts]: filtered, first map of the model, or default map.

File: src/app/app-interactive-map/store/interactive-map.effects.ts [hunk 3]
```diff
@@ -131,13 +151,9 @@ export class InteractiveMapEffects {
...
       const {maps} = storeState.interactiveMap;
-      const modelHasMaps = (mapItem: types.MapItem) =>
-        (mapItem.model === model.name);
-      const preferredMap = (mapItem: types.MapItem) =>
-        (mapItem.model === preferredMapItem.model && mapItem.name === preferredMapItem.name);
-      const predicate = (mapItem: types.MapItem) =>
-        (mapItem.model === model.name && mapItem.name === preferredMapsByModel[model.name]);
-      return new fromActions.SetMap(preferredSelector(predicate, preferredMap, modelHasMaps)(maps));
+      const preferred = maps.filter((item) => compareModel(item, preferredMapItem) && compareName(item, preferredMapItem));
+      const selectedMap = selectorCreator([compareModel, compareName])(maps, preferredMapsByModel[model.name] || {model: model.name}, preferred);
```
Comment (AlbaLopez) [inline src/app/app-interactive-map/store/interactive-map.effects.ts]: {model: model.name} for filter by model if the preferred doesn't exists for that model.

File: src/app/app-interactive-map/store/interactive-map.effects.ts [hunk 4]
```diff
@@ -131,13 +151,9 @@ export class InteractiveMapEffects {
...
+      return new fromActions.SetMap(selectedMap[0] || selectedMap);
```
Comment (AlbaLopez) [inline src/app/app-interactive-map/store/interactive-map.effects.ts]: to set the first one if is an array or just the element if it is an object.

**Pull request merged**